private/pkg/manifest/storage.go

```diff
@@ -102,9 +102,19 @@ func BucketWithNoExtraBlobsValidation() BucketOption {
 	}
 }
 
+// ReadBucket extends the storage.ReadBucket interface to add an accessor for the Manifest.
```

Comment:
This also feels like it breaks the abstraction of a bucket by providing direct data access

Reply:
This adds more metadata to a returned bucket that supports manifest/blobs. It isn't providing direct data access but just a separate way to represent a module.

Reply:
Removed this.